subsys/net/lib/azure_fota/azure_fota.c

```diff
@@ -82,10 +108,51 @@ static char current_version[CONFIG_AZURE_FOTA_VERSION_MAX_LEN] =
 BUILD_ASSERT(0, "CONFIG_AZURE_FOTA_SEC_TAG must be defined");
 #endif
 
+static void rep_status_set(enum fota_status new_status)
+{
+	current_status = new_status;
+}
+
+static void state_set(enum internal_state new_state)
+{
+	if (internal_state == new_state) {
+		return;
+	}
+
+	LOG_DBG("State transition: %s --> %s",
+		internal_state_str[internal_state],
+		internal_state_str[new_state]);
+
+	internal_state = new_state;
+}
+
+static enum internal_state state_get(void)
+{
+	return internal_state;
+}
+
+static char *reported_status_str_get(void)
+{
+	return rep_status_str[current_status];
+}
+
+static enum fota_status reported_status_get(const char *status)
+{
+	for (size_t i = 0; i < ARRAY_SIZE(rep_status_str); i++) {
+		if (strcmp(rep_status_str[i], status) == 0) {
+			return i;
+		}
+	}
+
+	return REP_STATUS_COUNT;
```

Comment:
Could we introduce an ```_INVALID``` status that is returned if the reported status is not present in the ```rep_status_str``` list? I guess the effect is the same, so no biggie.

Reply:
Yeah, initially had that, but removed as it's not strictly needed as you mention. But will re-introduce it for clarity